app/views/shared/_dark_mode_check.html.erb

```diff
@@ -1,24 +1,42 @@
 <script>
   // On page load or when changing themes, best to add inline in `head` to avoid FOUC
-  if (localStorage.theme === "dark" || (!("theme" in localStorage) && window.matchMedia("(prefers-color-scheme: dark)").matches)) {
-    document.documentElement.classList.add("dark");
-  } else {
-    document.documentElement.classList.remove("dark");
+  function isSystemTheme() {
+    return localStorage.getItem("theme") === "system";
   }
 
-  window.addEventListener("storage", function(event) {
-    if (this.localStorage.getItem("theme") === "light") {
-      if (document.documentElement.classList.contains("dark")) {
-        this.document.documentElement.classList.remove("dark");
-        this.document.querySelector("#app-darkmode-toggle>svg[data-dark-mode-target=darkIcon]").classList.remove("hidden");
-        this.document.querySelector("#app-darkmode-toggle>svg[data-dark-mode-target=lightIcon]").classList.add("hidden");
-      }
-    } else if (this.localStorage.getItem("theme") === "dark") {
-      if (!document.documentElement.classList.contains("dark")) {
-        this.document.documentElement.classList.add("dark");
-        this.document.querySelector("#app-darkmode-toggle>svg[data-dark-mode-target=darkIcon]").classList.add("hidden");
-        this.document.querySelector("#app-darkmode-toggle>svg[data-dark-mode-target=lightIcon]").classList.remove("hidden");
-      }
+  function isDarkTheme() {
+    return localStorage.getItem("theme") === "dark";
+  }
+
+  function isDarkMode() {
+    return (
+      isDarkTheme() ||
+      (isSystemTheme() &&
+        window.matchMedia("(prefers-color-scheme: dark)").matches)
+    );
+  }
+
+  function updateTheme(isDark) {
+    if (isDark) {
+      document.documentElement.classList.add("dark");
```

Comment:
**nitpick (blocking):** I think we should add the `dark` class but also remove the `light` class.

Reply:
Updated in 0599cbe